### PR TITLE
flutter: improve automatic instrumentation page

### PR DESCRIPTION
**Pull request by buenaflor**
Improves the automatic instrumentation page and the detail page for each instrumentation.

Ref #8564

### CODE REVIEW

File: src/platforms/flutter/integrations/asset-bundle-instrumentation.mdx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  'load',
+  bindToScope: true,
+);
+
+final text = await DefaultAssetBundle.of(context).loadString('assets/lorem-ipsum.txt');
+
+await transaction.finish(status: const SpanStatus.ok());
+```
+
+### 2. View the transaction on Sentry
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/asset-bundle-instrumentation.mdx:63]:
```suggestion
### 2. View the Transaction on Sentry.io
```

File: src/platforms/flutter/integrations/asset-bundle-instrumentation.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+To view the recorded transaction, log into [sentry.io](https://sentry.io) and open your project.
+Clicking **Performance** will open a page with transactions, where you can select the just recorded transaction with the name `asset-bundle-transaction`.
+
+## Additional Configuration
+
+### Disabling the `loadStructuredData` feature
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/asset-bundle-instrumentation.mdx:70]:
```suggestion
### Disabling the `loadStructuredData` Feature
```

File: src/platforms/flutter/integrations/app-start-instrumentation.mdx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  - mobile
+---
+
+Sentry's App Start Instrumentation provides insight into how long your application takes to launch.
+
+<Note>
+
+The App Start Instrumentation is shipped with Sentry's Flutter SDK automatically.
+
+</Note>
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/app-start-instrumentation.mdx:17]: Do we need this note? We don't have one on the Slow and Frozen Frames Instrumentation page, for example.

```suggestion
```

File: src/platforms/flutter/integrations/app-start-instrumentation.mdx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+## Instrumentation Behaviour
+
+Before diving into the configuration, it's important to understand how the App Start Instrumentation behaves:
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/app-start-instrumentation.mdx:21]:
```suggestion
Before diving into the configuration, it's important to understand how app start instrumentation behaves:
```

File: src/platforms/flutter/integrations/app-start-instrumentation.mdx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+- The SDK attaches the app start metrics to the **first run transaction**.
+- It tracks the length of time from the **earliest native process initialization** until the very first [PostFrameCallback](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/addPostFrameCallback.html) is triggered.
+- The SDK differentiates between a cold and a warm start, but doesn't track hot starts/resumes.
+
+## Prerequisite
+
+Before starting, ensure:
+
+1. Sentry Flutter SDK must be initialized. Learn more [here](/platforms/flutter/#configure)
+2. Performance Monitoring must be set up. Learn more [here](/platforms/flutter/performance/).
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/app-start-instrumentation.mdx:32]:
```suggestion
1. The Sentry Flutter SDK is nitialized. Learn more [here](/platforms/flutter/#configure)
2. Performance Monitoring is set up. Learn more [here](/platforms/flutter/performance/).
```

File: src/platforms/flutter/integrations/app-start-instrumentation.mdx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+## Configure
+
+The instrumentation is automatically enabled. There is no need for further configuration.
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/app-start-instrumentation.mdx:36]:
```suggestion
This type of instrumentation is automatically enabled. There is no need for further configuration.
```

File: src/platforms/flutter/integrations/app-start-instrumentation.mdx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import 'package:sentry/sentry.dart';
+
+final transaction = Sentry.startTransaction("test test", "my operation");
+await Future.delayed(const Duration(milliseconds: 1000));
+transaction.finish();
+```
+
+### 2. Locate Your Transaction:
+
+Open the Sentry performance dashboard, find and select the transaction you executed.
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/app-start-instrumentation.mdx:54]:
```suggestion
Open the [sentry.io](https://sentry.io/) performance dashboard, find, and select the transaction you executed.
```

File: src/platforms/flutter/integrations/app-start-instrumentation.mdx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+### 3. View App Start Metrics:
+
+Select the event within your transaction, Sentry displays the app start metrics on the right side of the screen.
```
Comment (shanamatthews) [inline src/platforms/flutter/integrations/app-start-instrumentation.mdx:58]:
```suggestion
Select the event within your transaction. Sentry.io displays the app start metrics on the right side of the screen.
```

File: src/platforms/dart/integrations/file.mdx [hunk 9]
```diff
@@ -6,17 +6,21 @@ redirect_from:
   - /platforms/dart/configuration/integrations/file/
 ---
 
-The `sentry_file` library provides [File](https://api.dart.dev/stable/2.18.5/dart-io/File-class.html) support for Sentry wrapping the `File` class with a `SentryFile` wrapper. The `SentryFile` is able to run performance monitoring for `File` operations.
+File I/O operations are fundamental for reading from and writing to files within an application.
+The `sentry_file` integration provides [File](https://api.dart.dev/stable/2.18.5/dart-io/File-class.html) support for Sentry thus providing insight into performance monitoring of file operations.
 
-The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/file/).
+## Behaviour
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/file.mdx:12]:
```suggestion
## Behavior
```

File: src/platforms/dart/integrations/file.mdx [hunk 10]
```diff
@@ -6,17 +6,21 @@ redirect_from:
...
+- When the operation throws an `Exception`, Sentry's SDK associates it with the running span.
 
-<Note>
+## Prerequisite
 
-The File I/O integration is only available for the `dart:io:File` class, not for the `dart:html:File` class.
-
-</Note>
+1. Sentry [Dart](/platforms/dart/#configure) SDK must be initialized.
+2. Performance Monitoring must be set up. Learn more [here](/platforms/dart/performance/).
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/file.mdx:23]:
```suggestion
1. The Sentry [Dart](/platforms/dart/#configure) SDK is initialized.
2. Performance Monitoring is set up. Learn more [here](/platforms/dart/performance/).
```

File: src/platforms/dart/integrations/file.mdx [hunk 11]
```diff
@@ -28,13 +32,17 @@ dependencies:
...
 
-Capturing transactions requires that you first <PlatformLink to="/performance/">set up performance monitoring</PlatformLink> if you haven't already.
+```dart
+final sentryFile = file.sentryTrace();
+```
 
-</Note>
+## Verify
+
+### 1. Execute the code
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/file.mdx:45]:
```suggestion
### 1. Execute the Code
```

File: src/platforms/dart/integrations/file.mdx [hunk 12]
```diff
@@ -83,4 +91,7 @@ Future<void> runApp() async {
 }
 ```
 
-To view the recorded transaction, log into [sentry.io](https://sentry.io) and open your project. Clicking **Performance** will open a page with transactions, where you can select the just recorded transaction with the name `MyFileExample`.
+### 2. View the recorded transaction on Sentry
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/file.mdx:94]:
```suggestion
### 2. View the Recorded Transaction on Sentry.io
```

File: src/platforms/dart/integrations/dio.mdx [hunk 13]
```diff
@@ -80,13 +78,26 @@ Future<void> main() async {
 
 ## Performance Monitoring for HTTP Requests
 
-<Note>
+The Dio integration also provides insight into performance monitoring for your HTTP requests done with Dio.
+
+### Instrumentation Behaviour
+
+- The created spans will be attached to the transaction on the scope - if no transaction is on the scope the Hive span will not be sent to Sentry.
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/dio.mdx:85]: Typo?

```suggestion
- The created spans will be attached to the transaction on the scope - if no transaction is on the scope the span will not be sent to Sentry.
```

File: src/platforms/dart/integrations/dio.mdx [hunk 14]
```diff
@@ -80,13 +78,26 @@ Future<void> main() async {
...
+- The span status depends on either the HTTP response code or `SpanStatus.internalError()` if the code does not match any of Sentry's SpanStatus options.
+- When the HTTP request throws an Exception, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and `SentryEvent` will be linked when viewing it on the Issue Details page in sentry.io.
+
+### Prerequisite
+
+Before starting, ensure:
 
-Capturing transactions requires that you first <PlatformLink to="/performance/">set up performance monitoring</PlatformLink> if you haven't already.
+1. Sentry Flutter SDK must be initialized. Learn more [here](/platforms/flutter/#configure)
+2. Performance Monitoring must be set up. Learn more [here](/platforms/flutter/performance/).
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/dio.mdx:96]:
```suggestion
1. The Sentry Flutter SDK is initialized. Learn more [here](/platforms/flutter/#configure).
2. Performance Monitoring is set up. Learn more [here](/platforms/flutter/performance/).
```

File: src/platforms/dart/integrations/dio.mdx [hunk 15]
```diff
@@ -95,3 +106,43 @@ final dio = Dio();
 
 dio.addSentry();
 ```
+
+### Verify
+
+#### 1. Execute the code
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/dio.mdx:112]:
```suggestion
#### 1. Execute the Code
```

File: src/platforms/dart/integrations/dio.mdx [hunk 16]
```diff
@@ -95,3 +106,43 @@ final dio = Dio();
...
+      span.throwable = exception;
+      span.status = const SpanStatus.internalError();
+      await Sentry.captureException(exception, stackTrace: stackTrace);
+    } finally {
+      await span.finish();
+    }
+}
+```
+
+#### 2. View the transaction on Sentry
```
Comment (shanamatthews) [inline src/platforms/dart/integrations/dio.mdx:145]:
```suggestion
#### 2. View the Transaction on Sentry.io
```

**Review by shanamatthews — APPROVED** [review]
Great job with these changes!

I ended up leaving quite a few comments, but it's all small stuff, like grammar and style guide fixes.

**Comment by buenaflor** [comment]
@shanamatthews thx so much for the review!

**Pull request merged**